### ISSUE TITLE
SCAN command falsly marks TYPE option as syntax error

### ISSUE THREAD

**Issue by 0xDiddi** [alicebob/miniredis#243]
[The redis docs on the `SCAN` command](https://redis.io/commands/scan) specify that a `TYPE` option can be supplied to return only objects of the given type.

After a wild goose chase brought about by transactions in the redis library I'm using masking underlying error messages (entirely unrelated, but it made getting to this root cause take two hours longer than it needed to), I eventually found that this is because the miniredis implementation of the `SCAN` command (see [cmd_generic.go:485](https://github.com/alicebob/miniredis/blob/master/cmd_generic.go#L485)) does not recognize the above mentioned `TYPE` option, and will give a syntax error when it is supplied.

Since I was going through the code anyway, I had a look what it might take to implement this option, and it doesn't look too bad. [The filter logic of `SCAN`](https://github.com/alicebob/miniredis/blob/master/cmd_generic.go#L531) should have access to [the key->type map](https://github.com/alicebob/miniredis/blob/master/miniredis.go#L40). Additionally, if I understand the redis docs correctly, the `TYPE` option only tests for equality and doesn't do any pattern matching.

I'll see if I can give that a shot in the next few days. Worst case if I can't get it to work would be to at least "eat" the option and not throw a syntax error, which (at least in my case) would stop the unit tests from exploding.

**Comment by alicebob** [comment]
Hi,
thank you for your issue!

> Redis version >= 6.0.0: Added the `TYPE` subcommand.

that explains why it's missing. A PR would be great, otherwise I'll have a look (just not today).

**Comment by 0xDiddi** [comment]
PR #248 has been opened which includes the logic for recognizing and filtering based on hte `TYPE` option, as well as unit and integration* tests for verifying it does so correctly.

The asterisk for integration tests is because I used a different version of redis (I was unable to use the `get_redis.sh` script due to some missing libraries, and brew only has 6.2.6 available), and additionally the tests panicking sometimes due to some connection error, and some integration tests unrelated to this PR were failing too.
**tl/dr** someone with a more reliable integration test setup should run those again.

**Comment by alicebob** [comment]
On Fri, Jan 14, 2022 at 10:50:15AM -0800, Kyaade M wrote:
> PR #248 has been opened which includes the logic for recognizing and filtering based on hte `TYPE` option, as well as unit and integration* tests for verifying it does so correctly.

thanks! I'll have a look soon, and will then see if and how the tests run here.
